### PR TITLE
New breaking API

### PR DESCRIPTION
**Pull request by torifat**
Please check the new [README](../blob/wip-next-version/README.md#better-ajv-errors)

**Changes**
- Using [`json-to-ast`](//npm.im/json-to-ast) to parse JSON
- Using [`@babel/code-frame`](//npm.im/@babel/code-frame) for pretty print
- Breaking API, check the [README](../blob/wip-next-version/README.md#better-ajv-errors)

**Actions**
- [x] Agree on the new API
- [x] @1999 Review the [typings.d.ts](/torifat/better-ajv-errors/blob/wip-next-version/README.md)
- [x] 🚢

---

I will add if I remember anything new 😄

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # better-ajv-errors
 
-JSON Schema validation for Human
+> JSON Schema validation for Human
```
[the remark this answers is not in the document]
Comment (torifat) [reply README.md]: You meant `blockquote`? Yes.

File: src/json/get-meta-from-path.js [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
   // TODO: Handle json pointer escape notation and better error handling
-  return getMeta(
-    dataPath
-      .split('/')
-      .slice(1)
-      .reduce((obj, pointer) => getValue(obj, pointer), parse(jsonString))
-  );
+  const pointers = dataPath.split('/').slice(1);
+  const lastPointerIndex = pointers.length - 1;
+  return pointers.reduce((obj, pointer, idx) => {
```
Comment (supertong) [inline src/json/get-meta-from-path.js]: Interesting... Didn't think of using `reduce` to travers through a tree before.. Learned.

Comment (torifat) [reply src/json/get-meta-from-path.js]: It looks clean 😄

Comment (supertong) [reply src/json/get-meta-from-path.js]: Yeah, love it!

File: src/json/get-meta-from-path.js [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
+    switch (obj.type) {
+      case 'Object': {
+        const filtered = obj.children.filter(
+          child => child.key.value === pointer
+        );
+        if (filtered.length !== 1) {
+          throw new Error(`Couldn't find property ${pointer} of ${dataPath}`);
+        }
+        const { key, value } = filtered[0];
+        return isIdentifierLocation && idx === lastPointerIndex ? key : value;
```
Comment (supertong) [inline src/json/get-meta-from-path.js]: Is this `isIdentifierLocation` determines whether the ☝️ is pointing at the `key` or the `value`?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (supertong) [reply README.md]: Ahh, cool. That's a `blockquote`

**Comment by supertong** [comment]
LGTM in general. Few small comments. 👍

File: src/json/get-meta-from-path.js (hunk 3 above)
Comment (supertong) [reply src/json/get-meta-from-path.js]: I see I see, just the variable name a bit confusing to me. Don't know what it means by just looking at it `isIdentifierLocation`.

Comment (torifat) [reply src/json/get-meta-from-path.js]: Yeah, you are right! I copied it from the caller `getLocation`. But, it doesn't make much in this _context_. Do you have any suggestion?

**Pull request merged**